Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -737,6 +739,24 @@ public static class Extensions {
             return (EnvDTE.DTE)provider.GetService(typeof(EnvDTE.DTE));
         }
 
+        internal static SVsShellDebugger GetShellDebugger(this IServiceProvider provider) {
+            return (SVsShellDebugger)provider.GetService(typeof(SVsShellDebugger));
+        }
+
+        internal static async System.Threading.Tasks.Task RefreshVariableViews(this IServiceProvider serviceProvider) {
+            EnvDTE.Debugger debugger = serviceProvider.GetDTE().Debugger;
+            foreach (AD7Engine engine in AD7Engine.GetEngines()) {
+                if (engine.Process != null && engine.Process.Id == debugger.CurrentProcess.ProcessID) {
```

Comment:
There might be more than one match if we're attached to a remote process that happens to have the same ID. But in this case the first match might not be the right one.

To get the exact instance, it needs to be comparing the transport qualifier (`Process2.TransportQualifier`) vs `PythonRemoteProcess.Uri` in addition to process ID. 

Alternatively, we can just refresh all active Python engines.